win10/RemoteBlinky/RemoteBlinky/ConnectionPage.xaml.cs

```diff
@@ -0,0 +1,338 @@
+                    break;
+
+                case "USB":
+
+                    //send telemetry about this connection attempt
+                    properties.Add( "Device_Name", device.Name );
+                    properties.Add( "Device_ID", device.Id );
+                    properties.Add( "Device_Kind", device.Kind.ToString() );
+                    App.Telemetry.TrackEvent( "USB_Connection_Attempt", properties );
+
+                    App.Connection = new UsbSerial( device );
+                    break;
+
+                case "Network":
+                    string host = NetworkHostNameTextBox.Text;
+                    string port = NetworkPortTextBox.Text;
+                    ushort portnum = 0;
+
+                    if( host == null || port == null )
+                    {
+                        ConnectMessage.Text = "You must enter host and IP.";
+                        return;
+                    }
+
+                    try
+                    {
+                        portnum = Convert.ToUInt16( port );
+                    }
+                    catch( FormatException )
+                    {
+                        ConnectMessage.Text = "You have entered an invalid port number.";
+                        return;
+                    }
+
+                    //send telemetry about this connection attempt
+                    properties.Add( "host", host );
+                    properties.Add( "port", portnum.ToString() );
+                    App.Telemetry.TrackEvent( "Network_Connection_Attempt", properties );
+
+                    App.Connection = new NetworkSerial( new Windows.Networking.HostName( host ), portnum );
+                    break;
+            }
+
+            App.Arduino = new RemoteDevice( App.Connection );
+            App.Arduino.DeviceReady += OnConnectionEstablished;
+            App.Arduino.DeviceConnectionFailed += OnConnectionFailed;
+
+            connectionAttemptStartedTime = DateTime.Now;
+            App.Connection.begin( 115200, SerialConfig.SERIAL_8N1 );
+
+            //start a timer for connection timeout
+            timeout = new DispatcherTimer();
+            timeout.Interval = new TimeSpan( 0, 0, 30 );
+            timeout.Tick += Connection_TimeOut;
+            timeout.Start();
+        }
+
+
+        /****************************************************************
+         *                  Event callbacks                             *
+         ****************************************************************/
+
+        private void OnConnectionFailed( string message )
+        {
+            var action = Dispatcher.RunAsync( Windows.UI.Core.CoreDispatcherPriority.Normal, new Windows.UI.Core.DispatchedHandler( () =>
+            {
+                timeout.Stop();
+
+                //telemetry
+                App.Telemetry.TrackRequest( "Connection_Failed_Event", DateTimeOffset.Now, DateTime.Now - connectionAttemptStartedTime, message, true );
```

Comment:
What's the consensus on using DateTime.Now as compared to DateTime.UtcNow?  My services experience is telling me not to use DateTime.Now.